components/ImageUpload/ImageUpload.js

```diff
@@ -85,12 +85,12 @@ class ImageUpload extends React.Component {
     const { id, email, first_name, last_name, description } = this.props.user
     const url = await firebase.storage().ref().child(`images/${this.state.id}`).getDownloadURL();
     if (this.props.currentComponent === 'User') {
-      this.props.setTempUserImage(null);
       this.props.setUserInfo({ description, email, first_name, id, last_name, image: url })
       const user = await apiCalls.patchUserImage(url, id)
     } else {
       this.props.setNewDogAddImage(url);
     }
+    this.props.setTempUserImage(null);
```

Comment:
Setting it here helps reset the correct image being set?

Reply:
I'm confused by this too...regardless of the conditional we want to set the temp user image to null?

Reply:
Yes, this clears the temporary user image so that it is not the same as a new dog image that has been added, it reverts back to the default image or a new one if it has been patched.